cloudify_terminal/tasks.py

```diff
@@ -32,8 +32,13 @@ def run(**kwargs):
     properties = ctx.node.properties
     terminal_auth = properties.get('terminal_auth', {})
     terminal_auth.update(kwargs.get('terminal_auth', {}))
+
     ip_list = terminal_auth.get('ip')
 
+    # testing against "None" value - when ip property in term_auth is wrongly set
+    if ip_list is None:
+        raise cfy_exc.NonRecoverableError("Property ip in terminal_auth is set to: %s" % str(ip_list))
+
```

Comment:
This doesn't look like the good fix.
If ip_list is`None` it can still be set if there is a host_ip property (in line 45). Therefore, failing before that means it will never arrive to the later code.
If there are users reaching a no-`host_id` property, the correct thing to do would explicitly check for the host_id property instead of checking vaguely the value of ip_list.
Also this begs the question why we are seeing no `host_id` properties. I would assume they are executing local workflows?